arcade/tests/core/test_catalog.py

```diff
@@ -136,3 +136,33 @@ def test_get_tool_by_name_with_invalid_version():
 
     with pytest.raises(ValueError):
         catalog.get_tool_by_name("SampleToolkit.SampleTool", version="2.0.0")
+
```

Comment:
Does it make sense to add a test to make sure this change does not introduce a regression? in other words, if `ARCADE_DISABLED_TOOLS` is _not_ set, things work normally

Reply:
I was thinking that the existing unit tests that add tools to catalog would verify there's no regression as long as they pass.

Or do you have a more involved regression test in mind that could give us more confidence?

Reply:
No I think you're right. The existing tests cover the regression angle.

Not sure if Python considers "unset" and "empty string" differently, in which case it would be useful to have a test that ensured `ARCADE_DISABLED_TOOLS=""` did not cause a crash.

Reply:
I like it. Added a couple tests
1. where `ARCADE_DISABLED_TOOLS=""`
2. where `ARCADE_DISABLED_TOOLS="      "`